src/library/scala/collection/immutable/Queue.scala

```diff
@@ -108,8 +108,12 @@ sealed class Queue[+A] protected(protected val in: List[A], protected val out: L
 
   override def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Queue[A], B, That]): That = {
     if (bf eq Queue.ReusableCBF) {
-      val thatQueue = that.asInstanceOf[Queue[B]]
-      new Queue[B](thatQueue.in ++ (thatQueue.out reverse_::: this.in), this.out).asInstanceOf[That]
+      if (that.isInstanceOf[Queue[_]]) {
```

Comment:
It would be clearer if you did something like `val newIn = if (that.isInstanceOf[Queue[_]]) ...` so you didn't have to repeat the `new Queue[B](_, this.out).asInstanceOf[That]` part.